File: designpatterns/22daily/adapter.py
class TemperatureType:
    Fahrenheit = 0
    Celsius = 1

class MeatDatabase:

    def GetSafeCookTemp(self, meat: str, tempType: TemperatureType):
        meat = meat.lower()
        safeTemp = None 
        if tempType == TemperatureType.Fahrenheit:
            safeTemp = 165
            if meat == "beef" or meat == "pork":
                safeTemp = 145
            elif meat == "chicken" or meat == "turkey":
                safeTemp = 165
        else:
            safeTemp = 74
            if meat == "beef" or meat == "pork" or meat == "veal":
                safeTemp = 63
            elif meat == "chicken" or meat == "turkey":
                safeTemp = 74
        return safeTemp

    def GetCaloriesPerOunce(self, meat: str):
        meat = meat.lower()
        calories = 0
        if meat == "beef":
            calories = 71
        elif meat == "pork":
            calories = 69
        elif meat == "chicken":
            calories = 66
        elif meat == "turkey":
            calories = 38
        return calories        

    def GetProteinPerOunce(self, meat: str):
        meat = meat.lower()
        protein = 0
        if meat == "beef":
            protein = 7.33
        elif meat == "pork":
            protein = 7.67
        elif meat == "chicken":
            protein = 8.57
        elif meat == "turkey":
            protein = 8.5
        return protein

# New code below, old (legacy) above
class Meat:

    def __init__(self, meat: str):
        self.meatName = meat

        self.safeCookTempFahrenheit = None
        self.safeCookTempCelsius = None
        self.caloriesPerOunce = None
        self.proteinPerOunce = None
        
    def LoadData(self):
        print(f"Meat: {self.meatName}")

class MeatDetails(Meat):

    def __init__(self, name):
        super().__init__(name)

    def LoadData(self):
        meatDatabase = MeatDatabase()

        self.safeCookTempFahrenheit = meatDatabase.GetSafeCookTemp(
            self.meatName,
            TemperatureType.Fahrenheit
        )
        self.safeCookTempCelsius = meatDatabase.GetSafeCookTemp(
            self.meatName,
            TemperatureType.Celsius
        )
        self.caloriesPerOunce = meatDatabase.GetCaloriesPerOunce(self.meatName)
        self.proteinPerOunce = meatDatabase.GetProteinPerOunce(self.meatName)

        super().LoadData()
        print(f"Safe Cook Temp (F): {self.safeCookTempFahrenheit}")
        print(f"Safe Cook Temp (C): {self.safeCookTempCelsius}")
        print(f"Calories per Ounce: {self.caloriesPerOunce}")
        print(f"Protein per Ounce: {self.proteinPerOunce}")

unknown = Meat("Beef")
unknown.LoadData()
print("---")

beef = MeatDetails("Beef")
beef.LoadData()
print("---")

turkey = MeatDetails("Turkey")
turkey.LoadData()
print("---")

chicken = MeatDetails("Chicken")
chicken.LoadData()


            

File: designpatterns/22daily/adapter/Adapter.cs
using System;
using System.Collections.Generic;

namespace adapter
{
    public enum TemperatureType
    {
        Fahrenheit,
        Celsius
    }

    // Legacy API
    class MeatDatabase
    {
        public float GetSafeCookTemp(string meat, TemperatureType tempType)
        {
            meat = meat.ToLower();
            if (tempType == TemperatureType.Fahrenheit)
                switch (meat)
                {
                    case "beef":
                    case "pork":
                    case "veal":
                        return 145f;
                    
                    case "chicken":
                    case "turkey":
                    default:
                        return 165f;
                }
            else if (tempType == TemperatureType.Celsius)
                switch (meat)
                {
                    case "beef":
                    case "pork":
                    case "veal":
                        return 64f;
                    
                    case "chicken":
                    case "turkey":
                    default:
                        return 74f;
                }
            throw new ArgumentException("Unknown temperature type: " + tempType.ToString());
        }

        public int GetCaloriesPer100g(string meat)
        {
            meat = meat.ToLower();

            switch (meat)
            {
                case "beef": return 250;
                case "veal": return 172;
                case "pork": return 242;
                case "chicken": return 239;
                case "turkey": return 189;
                default: return 0;
            }
        }

        public double GetProteinPer100g(string meat)
        {
            meat = meat.ToLower();

            switch (meat)
            {
                case "beef": return 26;
                case "veal": return 24;
                case "pork": return 21;
                case "chicken": return 27;
                case "turkey": return 29;
                default: return 0d;
            }
        }
    }

    class Meat
    {
        protected string name;
        protected Dictionary<TemperatureType, float> safeCookTemperature
            = new Dictionary<TemperatureType, float>();
        protected double caloredPer100g;
        protected double proteinPer100g;

        public Meat(string meat)
        {
            name = meat;
        }

        public virtual void LoadData()
        {
            Console.WriteLine("\nMeat: {0} ---------", name);
        }
    }

    class MeatDetails : Meat
    {
        private MeatDatabase meatDatabase;

        public MeatDetails(string name) : base(name) {}

        public override void LoadData()
        {
            meatDatabase = new MeatDatabase();

            safeCookTemperature[TemperatureType.Fahrenheit]
                = meatDatabase.GetSafeCookTemp(name, TemperatureType.Fahrenheit);
            safeCookTemperature[TemperatureType.Celsius]
                = meatDatabase.GetSafeCookTemp(name, TemperatureType.Celsius);

            caloredPer100g = meatDatabase.GetCaloriesPer100g(name);
            proteinPer100g = meatDatabase.GetProteinPer100g(name);

            base.LoadData();
            Console.WriteLine(
                " Safe Cook Temp (F): {0}",
                safeCookTemperature[TemperatureType.Fahrenheit]
            );
            Console.WriteLine(
                " Safe Cook Temp (C): {0}",
                safeCookTemperature[TemperatureType.Celsius]
            );
            Console.WriteLine(" Calories per 100g {0}", caloredPer100g);
            Console.WriteLine(" Protein per 100g {0}", proteinPer100g);
        }
    }
}